web/static/js/components/activity/ActivityDescription.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component, PropTypes } from 'react'
+import { translate } from 'react-i18next'
+
+class ActivityDescription extends Component {
+  text(activity) {
+    const { t } = this.props
+    const metadata = activity.metadata || {}
+    let reportType
+    switch (activity.entityType) {
+      case 'survey':
+        const surveyName = metadata['surveyName'] || 'Untitled'
+        switch (activity.action) {
+          case 'create':
+            return t('Created survey')
+          case 'rename':
+            const {oldSurveyName, newSurveyName} = metadata
+            if (oldSurveyName) {
+              return t('Renamed <i>{{oldSurveyName}}</i> survey to <i>{{newSurveyName}}</i>', {oldSurveyName: oldSurveyName, newSurveyName: newSurveyName})
+            } else {
+              return t('Named survey as <i>{{newSurveyName}}</i>', {newSurveyName: newSurveyName})
+            }
+          case 'edit':
+            return t('Edited <i>{{surveyName}}</i> survey', {surveyName: surveyName})
+          case 'delete':
+            return t('Deleted <i>{{surveyName}}</i> survey', {surveyName: surveyName})
+          case 'start':
+            return t('Started <i>{{surveyName}}</i> survey', {surveyName: surveyName})
+          case 'stop':
+            return t('Paused <i>{{surveyName}}</i> survey', {surveyName: surveyName})
+          case 'download':
+            reportType = metadata['reportType'].split('_').join(' ')
```

Comment:
This wont be translated if we construct the string like this. Please, extract this in a separate function using a  `switch` statement with each `reportType` possible value and return the name of the report tagged with `t`.